lib/utils/testem-events.js

```diff
@@ -157,27 +176,30 @@ class TestemEvents {
    * When all browsers complete, write test-execution.json to disk and clean up the stateManager
    *
    * @param {number} browserCount
-   * @param {number} browserid
+   * @param {number} launcherId
    * @param {Object} ui
-   * @param {boolean} loadBalance
-   * @param {string} fileName
-   * @param {string} writeExecutionFile
+   * @param {Object} commands
+   * @param {Object} currentDate
    */
-  completedBrowsersHandler(browserCount, browserId, ui, loadBalance, fileName, writeExecutionFile) {
+  completedBrowsersHandler(browserCount, launcherId, ui, commands, currentDate) {
     let browsersCompleted = false;
 
-    this.stateManager.incrementCompletedBrowsers(browserId);
+    this.stateManager.incrementCompletedBrowsers(launcherId);
     if (this.stateManager.getCompletedBrowser() === browserCount) {
-      if (writeExecutionFile && loadBalance) {
+      if (commands.get('writeModuleMetadataFile')) {
+        const moduleDetailFileName = path.join(this.root, `module-metadata-${currentDate}.json`);
+        const sortedModuleMetadata = getSortedModuleMetaData(this.stateManager.getModuleMetadata());
+
+        writeJsonToFile(moduleDetailFileName, sortedModuleMetadata, { spaces: 2 });
+        ui.writeLine(`\nThis execution module details was recorded at ${moduleDetailFileName}`);
+      }
+
+      if (commands.get('writeExecutionFile') && commands.get('loadBalance')) {
```

Comment:
Could this be its own function?

Reply:
Refactoring exam.js and testem-events.js will be in a separated pr as the current pr is getting bigger.